Signal/src/ViewControllers/AppSettingsViewController.m

```diff
@@ -238,6 +211,90 @@ - (void)updateTableContents
     self.contents = contents;
 }
 
+- (UITableViewCell *)profileHeaderCell
+{
+    UITableViewCell *cell = [UITableViewCell new];
+    cell.preservesSuperviewLayoutMargins = YES;
+    cell.contentView.preservesSuperviewLayoutMargins = YES;
+    cell.selectionStyle = UITableViewCellSelectionStyleNone;
+
+    const NSUInteger kAvatarSize = 68;
+    // TODO: Replace this icon.
+    UIImage *_Nullable localProfileAvatarImage = [OWSProfileManager.sharedManager localProfileAvatarImage];
+    UIImage *avatarImage = (localProfileAvatarImage
+            ?: [[UIImage imageNamed:@"profile_avatar_default"]
+                   imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]);
+    OWSAssert(avatarImage);
+
+    AvatarImageView *avatarView = [[AvatarImageView alloc] initWithImage:avatarImage];
+    if (!localProfileAvatarImage) {
+        avatarView.tintColor = [UIColor colorWithRGBHex:0x888888];
+    }
+    [cell.contentView addSubview:avatarView];
+    [avatarView autoVCenterInSuperview];
+    [avatarView autoPinLeadingToSuperView];
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+
+    if (!localProfileAvatarImage) {
+        UIImage *cameraImage = [UIImage imageNamed:@"settings-avatar-camera"];
+        cameraImage = [cameraImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+        UIImageView *cameraImageView = [[UIImageView alloc] initWithImage:cameraImage];
+        cameraImageView.tintColor = [UIColor ows_materialBlueColor];
+        [cell.contentView addSubview:cameraImageView];
+        [cameraImageView autoPinTrailingToView:avatarView];
+        [cameraImageView autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:avatarView];
+    }
+
+    UIView *threadNameView = [UIView containerView];
```

Comment:
I'd prefer to not have to know the origin of this copy and pasted code to be able to make sense of it. I'd say either give them reasonable enough names so they can stand on their own or extract something re-usable before you merge it in, in case the "I'll clean this up later" doesn't get priority for a while.